WC_Layers/OpenMC_PhotonTransport.py

```diff
@@ -0,0 +1,138 @@
+                 1.22e+6, 
+                 1.44e+6, 
+                 1.66e+6, 
+                 2.00e+6, 
+                 2.50e+6, 
+                 3.00e+6, 
+                 4.00e+6, 
+                 5.00e+6, 
+                 6.50e+6, 
+                 8.00e+6, 
+                 1.00e+7, 
+                 1.20e+7, 
+                 1.40e+7, 
+                 2.00e+7]
+
+#Define source:
+def make_source(W_Shell, C_Shell, Cells):
+    Source_List = []
+    Particle_Filter = openmc.ParticleFilter('photon')
+    Total_Mesh = openmc.UnstructuredMesh(Mesh_File, library='moab')
+    for index, bound in enumerate(bounds[:-1]):
+        Mesh_Dist = openmc.stats.MeshSpatial(Total_Mesh, strengths=strengths_list[index], volume_normalized=False)
+        Energy_Dist = openmc.stats.Uniform(a=bounds[index], b=bounds[index + 1])
+        #Source strengths given by strengths_list created by Source_Mesh_Reader
+        Source = Source_List.append(openmc.IndependentSource(space=Mesh_Dist, energy=Energy_Dist, strength=summed_strengths[index], particle='photon', domains=Cells))
+    return Source_List, Particle_Filter, Total_Mesh, Mesh_Dist
+
+# Define tallies
+def tallies(W_Shell, C_Shell, Particle_Filter, Total_Mesh):
+    Total_Filter = openmc.MeshFilter(Total_Mesh)
+    
+    neutron_tally = openmc.Tally(tally_id=1, name="Neutron tally")
+    neutron_tally.scores = ['flux', 'elastic', 'absorption']
+    
+    # Implementing filter for neutron tally through W shell
+    cell_filter = openmc.CellFilter([W_Shell, C_Shell])
+    neutron_tally.filters = [cell_filter, Total_Filter]
+
+    # Creating a tally to get the flux energy spectrum.
+    # An energy filter is created to assign to the flux tally.
+    energy_filter_flux = openmc.EnergyFilter.from_group_structure("VITAMIN-J-42")
+
+    spectrum_tally = openmc.Tally(tally_id=2, name="Flux spectrum")
+    # Implementing energy and cell filters for flux spectrum tally
+    spectrum_tally.filters = [cell_filter, Total_Filter, energy_filter_flux, Particle_Filter]
+    spectrum_tally.scores = ['flux']
+    
+    tall = openmc.Tallies([neutron_tally, spectrum_tally])
+    tall.export_to_xml()
+    return tall, neutron_tally, spectrum_tally, Total_Filter, cell_filter
+
+# Assign simulation settings
+def settings(Source_List):
+    sets = openmc.Settings()
+    sets.batches = 10
+    sets.inactive = 1
+    sets.particles = 100000
+    sets.source = Source_List
+    sets.run_mode = 'fixed source'
+    sets.export_to_xml()
+    return sets
+
+def plot_universe(Void, W_Shell, C_Shell, Cells):
+    universe_ss = openmc.Universe(cells=Cells)
+    Plot = universe_ss.plot(width=(2500.0, 2500.0), basis='xz',
+              colors={Void: 'blue', W_Shell: 'red', C_Shell: 'green'}, legend=True)
+    Plot.figure.savefig('Universe')
+    return universe_ss
+
+# Exporting materials, geometry, and tallies to .xml
+def export_to_xml(filepath, element_1, element_2, inner_radius_W, outer_radius_W, inner_radius_C):

```

Comment:
This function is doing too much - it should probably just be making an OpenMC model out of:
* materials
* geometry
* settings (incl. sourc)
* tallies

and returning that model

Reply:
I'd also recommend using lowercase/snakecase variable names